src/go/rpk/pkg/cli/topic/mount.go

```diff
@@ -0,0 +1,108 @@
+// by the Apache License, Version 2.0
+
+package topic
+
+import (
+	"time"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var alias string
+	var aliasNamespace string
+	var namespace string
+	var location string
+	var timeout int
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPICS...]",
+		Short: "Mount a topic to the Redpanda Cluster from tiered storage",
+		Long:  helpMount,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			topic := rpadmin.InboundTopic{
+				SourceTopic: rpadmin.Topic{
+					Topic:     topics[0],
+					Namespace: namespace,
+				},
+			}
+			if alias != "" && aliasNamespace != "" {
+				topic.Alias = &rpadmin.Topic{
+					Topic:     alias,
+					Namespace: aliasNamespace,
+				}
+			}
+			if location != "" {
+				topic.Location = location
+			}
+
+			migration, err := adm.AddInboundMigration(cmd.Context(), rpadmin.InboundMigration{
+				MigrationType:  "inbound",
+				Topics:         []rpadmin.InboundTopic{topic},
+				ConsumerGroups: []string{}, // not implemented yet
+			})
+			out.MaybeDie(err, "unable to add inbound migration: %v", err)
+
+			tmt := time.Duration(timeout * int(time.Minute/time.Nanosecond))
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.PrepareMigrationAction, "prepared", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to prepare migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.ExecuteMigrationAction, "executed", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to execute migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.FinishMigrationAction, "finished", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to finish migration", errP)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "Namespace for the mounted topic (optional)")
+	cmd.Flags().StringVarP(&location, "location", "l", "", "Location (optional)") // need more info on this
+	cmd.Flags().StringVarP(&alias, "alias", "a", "", "New name for the mounted topic, if different from the original (optional)")
+	cmd.Flags().StringVarP(&aliasNamespace, "aliasnamespace", "s", "", "Namespace for the alias topic, if using an alias (optional)")
+	cmd.Flags().IntVarP(&timeout, "timeout", "", 5, "Timeout in minutes for the migration to finish (default: 5)")
```

Comment:
Frankly I'm not even sure a timeout is warranted or a good idea here. Leaving a migration in a partially complete state because of an entirely arbitrary timeout seems less and less of a good idea the more I think about it so it might be sensible to simply run it forever and let the user cancel it manually if they have concerns about the state. 

It might also be sensible to work in a resume flag and functionality to support returning to an in progress migration should the context have been canceled. My read of the documentation available is that this is not incompatible with the state machine as designed.

Reply:
Oh and I removed the unwarranted default. I was erring on the side of caution as it has been a while since I've used cobra over urfave and I couldn't remember which elements it auto generated 😅